dib/build.go

```diff
@@ -71,7 +71,6 @@ func RebuildNode(node *dag.Node, builder types.ImageBuilder, testRunners []types
 		err := doRebuild(img, builder, rateLimiter, newTag, localOnly)
 		if err != nil {
 			reportChan <- report.withError(err)
-			close(reportChan)
```

Comment:
This means the rebuild process will continue until all the images are processed. But when the build fails for an image, its children cannot be rebuilt, so the goroutine will be struck and wait forever.
You have to do something to prevent this situation

Reply:
This can maybe be fixed by adding a new flag `RebuildFailed` and set it to `true`, so the children can skip the build and return immediately if they find a parent with the flag to `true`